haystack/preview/components/routers/document_joiner.py

```diff
@@ -0,0 +1,142 @@
+import itertools
+import logging
+from collections import defaultdict
+from math import inf
+from typing import List, Optional
+from canals.component.types import Variadic
+
+from haystack.preview import component, Document
+
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class DocumentJoiner:
+    """
+    A component that joins input lists of documents from multiple connections and outputs them as one list.
+
+    The component allows multiple join modes:
+    * concatenate: combine the documents from multiple nodes. Any duplicate documents are discarded.
+                   The score is only determined by the last node that outputs the document.
```

Comment:
This is a bit unclear. Does it mean the score is determined by the node from which DocumentJoiner gets the documents? or by the last node in the pipeline? what if that node doesn't assign a score?

Maybe something like: Documents get their scores from the last component in the pipeline that assigns scores. This join mode doesn't influence Document scores.